Assets/MixedRealityToolkit/MSBuild/Scripts/CSProjectInfo.cs

```diff
@@ -0,0 +1,363 @@
+                foreach (string source in Assembly.sourceFiles)
+                {
+                    ProcessSourceFile(source, sourceIncludeTemplate, sourceIncludes, sourceGuidToClassName);
+                }
+
+                File.WriteAllLines(Path.Combine(propsOutputFolder, $"{Guid.ToString()}.csmap"), sourceGuidToClassName.Select(t => $"{t.Key}:{t.Value}"));
+
+                bool bothConfigurations = PlayerPlatforms.Count > 0;
+
+                List<string> supportedPlatformBuildConditions = new List<string>();
+                PopulateSupportedPlatformBuildConditions(supportedPlatformBuildConditions, suportedPlatformBuildConditionTemplate, "InEditor", InEditorPlatforms);
+                PopulateSupportedPlatformBuildConditions(supportedPlatformBuildConditions, suportedPlatformBuildConditionTemplate, "Player", PlayerPlatforms);
+
+                HashSet<string> inEditorSearchPaths = new HashSet<string>(), playerSearchPaths = new HashSet<string>();
+                string projectReferences = string.Join("\r\n", CreateProjectReferencesSet(projectReferenceSetTemplate, inEditorSearchPaths, true), CreateProjectReferencesSet(projectReferenceSetTemplate, playerSearchPaths, false));
+                Dictionary<string, string> tokens = new Dictionary<string, string>()
+                {
+                    { "<!--PROJECT_GUID_TOKEN-->", Guid.ToString() },
+                    { "<!--ALLOW_UNSAFE_TOKEN-->", Assembly.compilerOptions.AllowUnsafeCode.ToString() },
+                    { "<!--LANGUAGE_VERSION_TOKEN-->", MSBuildTools.CSharpVersion },
+
+                    { "<!--DEVELOPMENT_BUILD_TOKEN-->", "false" }, // Default to false
+
+                    { "<!--OUTPUT_PATH_TOKEN-->", Path.Combine("..", "MRTKBuild") },
+                    { "<!--IS_EDITOR_ONLY_TARGET_TOKEN-->", (ProjectType ==  ProjectType.EditorAsmDef || ProjectType == ProjectType.PredefinedEditorAssembly).ToString() },
+                    { "<!--UNITY_EDITOR_INSTALL_FOLDER-->", Path.GetDirectoryName(EditorApplication.applicationPath) + "\\"},
+
+                    { "<!--DEFAULT_PLATFORM_TOKEN-->", availablePlatforms.First(t=>t.BuildTarget == BuildTarget.StandaloneWindows).Name },
+
+                    { "<!--SUPPORTED_PLATFORMS_TOKEN-->", string.Join(";", availablePlatforms.Select(t=>t.Name)) },
+
+                    { "<!--INEDITOR_ASSEMBLY_SEARCH_PATHS_TOKEN-->", string.Join(";", inEditorSearchPaths) },
+                    { "<!--PLAYER_ASSEMBLY_SEARCH_PATHS_TOKEN-->", string.Join(";", playerSearchPaths) },
+
+                    { "##PLATFORM_PROPS_FOLDER_PATH_TOKEN##", propsOutputFolder },
+
+                    { projectReferenceSetTemplate, projectReferences },
+                    { sourceIncludeTemplate, string.Join("\r\n", sourceIncludes) },
+                    { suportedPlatformBuildConditionTemplate, string.Join("\r\n", supportedPlatformBuildConditions) }
```

Comment:
Are any of these optional? What I'm curious about is if these templates end up getting validated in making sure that they have the required tokens. i.e. it's pretty easy for me to make a mispelling in a token, and if so, substitution here would silently fail (even though I would expect that tokens should be validated as being valid and present)